bin/sandman/sweep.py

```diff
@@ -120,41 +128,101 @@ def _files(
+                    },
+                    "staged": {
+                        file.path: {"filepath": file.path}
+                        for file in _files(State.Staged)
+                    }
+                }
+
+                n_files = {
+                    "deleted": len(file_lists["deleted"]),
+                    "staged": len(file_lists["staged"]),
+                }
+
+                tminuses: T.Dict[str, int] = {}
+                lustre_paths: T.Dict[str, Path] = {}
+
+                _email_constructors: T.Dict[str, T.Type[MessageNamespace.Message]] = {
+                    "deleted": MessageNamespace.DeletedEmail,
+                    "staged": MessageNamespace.StagedEmail
+                }
 
                 # Warned files that require notification
-                for tminus in (*config.deletion.warnings,
-                               config.sandman_run_interval):
-                    to_warn = _files(State.Warned, tminus=tminus)
+                _warn_type = "urgent"
+                for tminus in (config.sandman_run_interval, *config.deletion.warnings):
 
+                    to_warn = _files(State.Warned, tminus=tminus)
                     hours = int(time.seconds(tminus) / 3600)
-                    attachments[f"delete-{hours}"] = to_warn
-                    warned.append((tminus, to_warn.accumulator))
-
-                # Construct e-mail and add non-trivial attachments
-                non_trivial = False
-                mail = NotificationEMail(stakeholder, deleted, staged, warned)
-                for filename, files in attachments.items():
-                    if len(files) > 0:
-                        non_trivial = True
-                        mail += GZippedFOFN(f"{filename}.fofn.gz",
-                                            [file.path for file in files])
-                if non_trivial:
+
+                    _key = f"delete-{hours}" if _warn_type != "urgent" else _warn_type
+                    attachments[_key] = to_warn
+                    group_summaries[_key] = to_warn.accumulator
+                    file_lists[_key] = {
+                        file.path: {"filepath": file.path} for file in to_warn
+                    }
+                    n_files[_key] = len(
+                        file_lists[_key])
+                    tminuses[_key] = time.seconds(tminus)
+                    lustre_paths[_key] = Path("")  # TODO
+                    _email_constructors[_key] = MessageNamespace.WarnedEmail if _warn_type != "urgent" else MessageNamespace.UrgentEmail
+
+                    # first one is urgent email, so now we can set it to "warning"
+                    _warn_type = "warning"
+
+                for key, cons in _email_constructors.items():
+                    if not attachments[key]:
+                        log.debug(
+                            f"Skipping E-mail for {key} files to {stakeholder.name} ({stakeholder.email}): Trivial e-mail")
```

Comment:
I removed this on purpose; is there any reason to be logging the fact that we got in to this loop with no files? Just say nothing?

Reply:
removed it again - would have been lost in some conflict resolving somewhere